packages/vm/src/imports.rs

```diff
@@ -1606,14 +1619,14 @@ mod tests {
 
         let msg = vec![0x22; MESSAGE_HASH_MAX_LEN];
         let msg_ptr = write_data(&env, &msg);
-        let sig = vec![0x22; EDDSA_SIGNATURE_LEN];
+        let sig = vec![0x22; MAX_LENGTH_ED25519_SIGNATURE];
         let sig_ptr = write_data(&env, &sig);
         let pubkey = vec![0x04; EDDSA_PUBKEY_LEN];
         let pubkey_ptr = write_data(&env, &pubkey);
 
         assert_eq!(
             do_ed25519_verify::<MA, MS, MQ>(&env, msg_ptr, sig_ptr, pubkey_ptr).unwrap(),
-            10 // mapped GenericErr
+            1 // verification failure
```

Comment:
How was this passing? I guess `ed25519_verify` internal handling of errors changed.

Reply:
So the change is the following: In Ed25519 for the pubkey format it is only checked that we get 32 bytes. Everything passing this check ends up as a verification failure. This avoid different bahavior for pubkeys depending on wheter or not they are points on a curve or not. This gives us the same bahaviour as in the badge case, where every [u8; 32] can be converted to an input for the batch and all further checks result in invalid verification.

Reply:
Thanks.